core/blockchain.go

```diff
@@ -282,6 +284,7 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, chainConfig *par
 	}
 	// Take ownership of this particular state
 	go bc.update()
+	go bc.updateTxIndices()
```

Comment:
I think we can be a bit more reliable here, but I'm not sure whether we want to or not.

The current code is fully background: if I specify a smaller `tx limit`, it will slowly crunch in the background and delete unneeded stuff. This is desirable.

Pondering about the other way though. If I ran Geth with a limit of say 30K, and realize it's too little and change to 300K, should we do the expansion on a background thread, or should we "fix the database" on startup? Doing in the the background would be "nicer", doing it in the foreground would be more predictable.

My main concern is that we need a way to programatically decide if the node is ready or not. If we block on startup with indexing, it's obvious (APIs are not opened until indexinig finished), but it's overkill for users. Maybe a good alternative would be to keep the background approach, but have an `eth.indexing` API, the same way we have `eth.syncing`. We could add stats about tx indexing, bloombits, cht, etc.

Reply:
Probably we need a way to disable this goroutine in certain cases where we don't want background updates happening (geth init, geth inspect, etc).